payments-core/src/main/java/com/stripe/android/view/StripeEditText.kt

```diff
@@ -196,6 +196,17 @@ open class StripeEditText @JvmOverloads constructor(
     }
 
     private fun listenForDeleteEmpty() {
+        // On some devices, the OnKeyListener isn't invoked for software keyboards. It is invoked on
+        // other devices such as my Pixel. To fix the issue for all devices, we're adding an
+        // additional text watcher to keep isLastKeyDelete in the correct state.
+        val textWatcher = object : StripeTextWatcher() {
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                isLastKeyDelete = count == 0
+            }
+        }
+
+        addTextChangedListener(textWatcher)
```

Comment:
Thinking out loud: `listenForDeleteEmpty()` is only called once in the constructor, so this is safe. If this constraint were to change, we wouldn't want to add multiple text watchers.

Reply:
Right, I added a check to make sure we only ever add this once. Just in case this is ever moved anywhere else.